File: Two_Pointers/Patterns/Pattern8/Pattern8.py
target = 7
nums = [2,3,1,2,4,3,7,9]
left=0
right=0
summ=0
minm=123456789

while right<len(nums) and right>=left:
    summ+=nums[right]
    while summ>=target:
        minm=min(minm,right-left+1)
        summ-=nums[left]
        left+=1
    right+=1
print(minm)

File: Two_Pointers/Patterns/Pattern8/Pattern8.java
import java.math.BigInteger;

class Pattern8 {
    public static void main(String[] args) {
        int minn=Integer.MAX_VALUE;
        int sum=0;
        int right=0;
        int left=0;
        int[] nums = {2,3,1,2,1,3};
        int target=70;

        while(right<nums.length){
            sum+=nums[right];

            while(sum>=target){
                minn=Math.min(right-left+1,minn);
                sum-=nums[left];
                left++;
            }
            right++;
        }
        if(minn==Integer.MAX_VALUE){
            System.out.println("0");
        }
        else{
            System.out.println(minn);
        }
    }
}